src/components/Countdown/Countdown.tsx

```diff
@@ -0,0 +1,151 @@
+import { ComponentProps, forwardRef, useCallback, useEffect, useState } from 'react';
+import { getCommonProps } from '../../utils';
+import classnames from 'classnames';
+
+// You'll need to change the ComponentProps<"htmlelementname"> to match the top-level element of your component
+export interface CountdownProps extends ComponentProps<'div'> {
+  /**
+   * The date the countdown should end
+   */
+  endDate: string;
+  /**
+   * Any descriptor to appear below the coundown
+   */
+  intervalDescription?: string;
+  /**
+   * Text translation for days
+   */
+  daysLabel?: string;
+  /**
+   * Text translation for hours
+   */
+  hoursLabel?: string;
+  /**
+   * Text translation for minutes
+   */
+  minutesLabel?: string;
+  /**
+   * Text translation for seconds
+   */
+  secondsLabel?: string;
+  /**
+   * Text translation for what the countdown is for
+   */
+  label?: string;
+}
+/**
+ * ## Overview
+ *
+ * Overview of this widget
+ *
+ * [Figma Link](https://www.figma.com/design/hMu9IWH5N3KamJy8tLFdyV/EASEL-Compendium%3A-Tokens%2C-Components-%26-Patterns?node-id=10850-35410&m=dev)
+ *
+ * [Storybook Link](https://phillips-seldon.netlify.app/?path=/docs/components-countdown--overview)
+ */
+const Countdown = forwardRef<HTMLDivElement, CountdownProps>(
+  (
+    {
+      endDate,
+      label = 'Lots Close in',
+      daysLabel = 'Days',
+      hoursLabel = 'Hours',
+      minutesLabel = 'Minutes',
+      secondsLabel = 'Seconds',
+      intervalDescription = 'Lots Close in 2-minute intervals',
+      className,
+      ...props
+    },
+    ref,
+  ) => {
+    const { className: baseClassName, ...commonProps } = getCommonProps(props, 'Countdown');
+
+    const calculateTimeLeft = useCallback(() => {
+      const targetTime = new Date(endDate).getTime();
+      const currentTime = new Date().getTime();
+      const difference = targetTime - currentTime;
+
+      // Adjust for GMT offsets and Daylight Saving Time
+      const targetDate = new Date(endDate);
+      const currentDate = new Date();
```

Comment:
For me, rather than have `timeLeft` be the state, it makes more sense to store `currentDateTime`, `setCurrentDateTime` as state, then have the `date-fns` calculate the time lefts from that.